File: Good-First-Issues/tell_triangle_type.cpp
#include <iostream>
#include <cmath>
using namespace std;
 
// Function to check if the triangle
void checkTriangle(int x, int y, int z)
{
	// sorthing x, y and z.
    if (x > z){
    	x = x + z;
		z = x - z;
		x = x - z;
    }

	if (x > y){
		x = x + y;
		y = x - y;
		x = x - y;
	}
	//Now the smallest element is the 1st one. Just check the 2nd and 3rd
	if (y > z){
		y = y + z;
		z = y - z;
		y = y - z;
	}

	// Check for triangle valid or not.
	if (x + y <= z) {
        cout << "Triangle not possible";
    	return;
    }

    // Check for equilateral triangle
    else if (x == y && y == z)
        cout << "Equilateral";
 
    // Check for isosceles triangle
    else if (x == y || y == z || z == x)
        cout << "Isosceles";
 
    // Otherwise scalene triangle
    else {
        cout << "Scalen" << ", ";

        // Check for Right angled triangle
	    if (pow(z,2) == pow(x, 2) + pow(y, 2))
	        cout << "Right-angled";

	    // Check for Obtuse angle triangle
	    else if(pow(z,2) > pow(x, 2) + pow(y, 2))
	        cout << "Obtuse-angled";
	    
	    else
	        cout << "Acute-angled";
	}
}

// Driver Code
int main()
{    
    // Given sides of triangle
    int x , y , z;
    cout << "Enter length of side of triangle : ";
    cin >> x >> y >> z;
     
    // Function call
    checkTriangle(x, y, z);
    cout << endl;
}

File: Good-First-Issues/Tell_Triangle_type.py
def check_triangle(a,b,c):
    result = True
    # function to check whether given sides makes a triangle or not
    if a+b <= c or b+c <= a or c+a <= b :
        result = False
    return result

def type_side(a,b,c):
    # Function to determine whether a triangle is equilateral or isosceles or scalene
    if a == b == c :
        ans = "Equilateral"
    elif a==b or b==c or c==a:
        ans = "Isosceles"
    else:
        ans = "Scalene"
    return ans

def type_angle(p,q,r):
    # Function to determine whether a triangle is right or acute or obtuse angled 
    if p**2 + q**2 == r**2:
        ans = "Right Angled Triangle"
    elif p**2 + q**2 > r**2:
        ans = "Actue Angled Triangle"
    else:
        ans = "Obtuse Angled Triangle"
    return ans

#driver code
a,b,c = [int(i) for i in input().split()]
if check_triangle(a,b,c):
    aa = type_side(a,b,c)
    arr = [a,b,c]
    r = max(arr)
    p,q = [j for j in arr if j!=r]
    bb = type_angle(p,q,r)
    print(aa,",",bb)
else:
    print("Triangle not possible")
